fem/linearform.cpp

```diff
@@ -155,18 +157,32 @@ bool LinearForm::SupportsDevice()
    return true;
 }
 
-void LinearForm::Assemble(bool use_device)
+void LinearForm::SetAssemblyLevel(AssemblyLevel assembly_level)
+{
+   if (ext)
+   {
+      MFEM_ABORT("the assembly level has already been set!");
+   }
+   assembly = assembly_level;
+   switch (assembly)
+   {
+      case AssemblyLevel::LEGACY:
```

Comment:
I'm wondering if it would be possible to allow the user to switch back to `AssemblyLevel::LEGACY`. Would we just need to delete `ext` if it exists? If it is that simple then I think we should do it.

Reply:
This mirrors the behavior of BilinearForm in https://github.com/mfem/mfem/blob/master/fem/bilinearform.cpp#L115-L141.

Currently we do not allow changing the assembly level in neither, probably because that is potentially a very expensive operation. 

We can consider doing that, but maybe that should be a separate PR?

What do you think @camierjs, @pazner, @v-dobrev?